From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nostalfinals <yuu8583@proton.me>
Date: Tue, 21 Apr 2026 09:07:11 +0800
Subject: [PATCH] Skip firing PreCreatureSpawnEvent when there are no listeners


diff --git a/net/minecraft/world/level/NaturalSpawner.java b/net/minecraft/world/level/NaturalSpawner.java
index 4e5078495037078922ab6350a500acdabe177ca5..4dd03762cfc37d4a0565c7e2f39e411e9f703eb6 100644
--- a/net/minecraft/world/level/NaturalSpawner.java
+++ b/net/minecraft/world/level/NaturalSpawner.java
@@ -367,6 +367,8 @@ public final class NaturalSpawner {
     ) {
         EntityType<?> entityType = data.type();
         // Paper start - PreCreatureSpawnEvent
+        // Nix start - Skip firing PreCreatureSpawnEvent when there are no listeners
+        if (com.destroystokyo.paper.event.entity.PreCreatureSpawnEvent.getHandlerList().getRegisteredListeners().length > 0) {
         com.destroystokyo.paper.event.entity.PreCreatureSpawnEvent event = new com.destroystokyo.paper.event.entity.PreCreatureSpawnEvent(
                 org.bukkit.craftbukkit.util.CraftLocation.toBukkit(pos, level),
                 org.bukkit.craftbukkit.entity.CraftEntityType.minecraftToBukkit(entityType), org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.NATURAL
@@ -377,6 +379,8 @@ public final class NaturalSpawner {
             }
             return PreSpawnStatus.CANCELLED;
         }
+        }
+        // Nix end - Skip firing PreCreatureSpawnEvent when there are no listeners
         final boolean success = entityType.getCategory() != MobCategory.MISC
         // Paper end - PreCreatureSpawnEvent
             && (
