From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nostalfinals <yuu8583@proton.me>
Date: Mon, 20 Apr 2026 23:12:58 +0800
Subject: [PATCH] Rebrand


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index 4e099b584c3435ba1ebce492011d35ae1ac0cdf8..d1f7090277ab300bcb25d5f2d9ac71e0ce1042c1 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -17,6 +17,13 @@ public interface ServerBuildInfo {
      */
     Key BRAND_PAPER_ID = Key.key("papermc", "paper");
 
+    // Nix start - Rebrand
+    /**
+     * The brand id for Nix.
+     */
+    Key BRAND_NIX_ID = Key.key("plutoproject", "nix");
+    // Nix end - Rebrand
+
     /**
      * Gets the {@code ServerBuildInfo}.
      *
